### ISSUE TITLE
qiskit_algorithms namespace is not supported.

### ISSUE THREAD

**Issue by airwoodix** [qiskit-community/qiskit-optimization#535]
### Environment

- **Qiskit Optimization version**: 0.5.0
- **Python version**: 3.11
- **Operating system**: Linux

### What is happening?

`MinimumEigenOptimizer` doesn't support algorithms from `qiskit-algorithms`.

The `qiskit.algorithms` namespace is [deprecated](https://qiskit.org/documentation/apidoc/algorithms.html) since `qiskit-terra` v0.25.0. The `qiskit_algorithms` namespace from [`qiskit-algorithms`](https://pypi.org/project/qiskit-algorithms/) should be a drop-in replacement (for the transition period).

### How can we reproduce the issue?

For example:

```python
from qiskit_algorithms.minimum_eigensolvers import QAOA
from qiskit_optimization.algorithms import MinimumEigenOptimizer

_ = MinimumEigenOptimizer(min_eigen_solver=QAOA(...))
```

fails with:

```
TypeError: MinimumEigenOptimizer supports qiskit.algorithms.minimum_eigensolvers.SamplingMinimumEigensolver, qiskit.algorithms.minimum_eigensolvers.NumPyMinimumEigensolver, and qiskit.algorithms.minimum_eigen_solvers.MinimumEigensolver. But <class 'qiskit_algorithms.minimum_eigensolvers.qaoa.QAOA'> is given.
```

raised at https://github.com/qiskit-community/qiskit-optimization/blob/5e8bc0fe22f9f28569eca95ceed20db8c8990f61/qiskit_optimization/algorithms/minimum_eigen_optimizer.py#L164

### What should happen?

It should be possible to use legacy `qiskit.algorithms` and `qiskit_algorithms` namespaces interchangeably (until the latter diverges).

### Any suggestions?

To ease up the transition and keep the type annotations correct, one could prefer `qiskit_algorithms` if it is present:

```python
try:
    from qiskit_algorithms.minimum_eigensolvers import ...
except ImportError:
    from qiskit.algorithms.minimum_eigensolvers import ...
)
```

The `TypeError` message should already now be informative enough to figure out what is happening.

**Comment by t-imamichi** [comment]
It has been resolved by #537

**Comment by airwoodix** [comment]
Awesome, thanks!
